invokeai/backend/model_manager/load/load_base.py

```diff
@@ -35,6 +40,16 @@ def __exit__(self, *args: Any, **kwargs: Any) -> None:
         """Context exit."""
         self._locker.unlock()
 
+    @contextmanager
+    def model_on_device(self) -> Generator[Tuple[Optional[Dict[str, torch.Tensor]], AnyModel], None, None]:
+        """Return a tuple consisting of the model's state dict (if it exists) and the locked model on execution device."""
+        try:
+            state_dict = self._locker.get_state_dict()
+            locked_model = self._locker.lock()
+            yield (state_dict, locked_model)
+        finally:
+            self._locker.unlock()
```

Comment:
Only unlock if we successfully locked.
```suggestion
        locked_model = self._locker.lock()
        try:
            state_dict = self._locker.get_state_dict()
            yield (state_dict, locked_model)
        finally:
            self._locker.unlock()
```